File: ainpc-api/src/main/java/ro/ainpc/api/AddonRegistryApi.java
package ro.ainpc.api;

import ro.ainpc.addons.AINPCAddon;
import ro.ainpc.addons.AddonDescriptor;
import ro.ainpc.addons.AddonType;

import java.util.Collection;
import java.util.List;

public interface AddonRegistryApi {

    void registerDescriptor(AddonDescriptor descriptor);

    void registerAddon(AINPCAddon addon);

    void unregisterAddon(String addonId);

    void removeByOrigin(String origin);

    Collection<AddonDescriptor> getDescriptors();

    List<AddonDescriptor> getDescriptors(AddonType type);

    AddonDescriptor getDescriptor(String id);

    AddonDescriptor getPrimaryScenario();

    default boolean isAddonEnabled(String addonId) {
        return true;
    }

    int size();
}

File: ainpc-api/src/main/kotlin/ro/ainpc/api/AddonRegistryApi.kt
package ro.ainpc.api

import ro.ainpc.addons.AINPCAddon
import ro.ainpc.addons.AddonDescriptor
import ro.ainpc.addons.AddonType

interface AddonRegistryApi {
    val descriptors: Collection<AddonDescriptor>

    val primaryScenario: AddonDescriptor?

    fun registerDescriptor(descriptor: AddonDescriptor?)

    fun registerAddon(addon: AINPCAddon?)

    fun unregisterAddon(addonId: String?)

    fun removeByOrigin(origin: String?)

    fun getDescriptors(type: AddonType?): List<AddonDescriptor>

    fun getDescriptor(id: String?): AddonDescriptor?

    fun isAddonEnabled(addonId: String?): Boolean = true

    fun size(): Int
}
